pantry/src/pantry.rs

```diff
@@ -326,17 +332,34 @@ impl PantryEntry {
                 block_size,
             );
         }
+        if Self::MAX_CHUNK_SIZE % block_size as usize != 0 {
+            crucible_bail!(
+                InvalidNumberOfBlocks,
```

Comment:
Maybe should be

```rust
    #[error("Data length not block size multiple")]
    DataLenUnaligned,
```

?

Reply:
I added static assertions relating `MAX_CHUNK_SIZE` and `MAX_BLOCK_SIZE`, so this should always be true and I switched it to an assertion (4c2c6575e5528aa4974ebc4dd8d0237e357e9594)